OpenMarket/OpenMarket/Controller/ProductListViewController.swift

```diff
@@ -0,0 +1,178 @@
+    
+    case main
+}
+
+final class ProductListViewController: UIViewController {
+    
+    var collectionView: UICollectionView!
+    private var dataSource: UICollectionViewDiffableDataSource<Section, Product>!
+    var productData: ProductList?
+    
+    override func viewDidLoad() {
+        super.viewDidLoad()
+        
+        view.backgroundColor = .white
+        configureSegmentedControl()
+        configureNavigationBar()
+        configureAddButton()
+        
+        let networkProvider = NetworkAPIProvider()
+        
+        networkProvider.fetchProductList(query: [.itemsPerPage: "200"]) { [weak self] result in
+            switch result {
+            case .success(let data):
+                self?.productData = data
+                DispatchQueue.main.async {
+                    self?.configureHierarchy()
+                    self?.configureDataSource()
+                }
+            default :
+                return
+            }
+        }
+    }
+    
+    @objc private func addButtonPressed() {
+        let addProductViewController = AddProductViewController()
+        self.present(addProductViewController, animated: true, completion: nil)
+    }
+}
+
+extension ProductListViewController {
+    
+    private func configureNavigationBar() {
+        let navigationBarAppearance = UINavigationBarAppearance()
+        navigationBarAppearance.backgroundColor = .systemGray6
+        self.navigationController?.navigationBar.scrollEdgeAppearance = navigationBarAppearance
+    }
+    
+    private func configureSegmentedControl() {
+        let segmentTextContent = [NSLocalizedString("LIST", comment: ""),
+                                  NSLocalizedString("GRID", comment: "")]
+        let segmentedControl = UISegmentedControl(items: segmentTextContent)
+        segmentedControl.selectedSegmentIndex = 0
+        segmentedControl.backgroundColor = .systemBackground
+        segmentedControl.selectedSegmentTintColor = .systemBlue
+        segmentedControl.frame = CGRect(x: 0, y: 0, width: 200, height: 30)
+        segmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.white],
+                                                for: .selected)
+        segmentedControl.setTitleTextAttributes([NSAttributedString.Key.foregroundColor: UIColor.systemBlue],
+                                                for: .normal)
+        segmentedControl.addTarget(self, action: #selector(segControlChanged),
+                                   for: UIControl.Event.valueChanged)
+        self.navigationItem.titleView = segmentedControl
+    }
+    
+    @objc func segControlChanged(segcon: UISegmentedControl) {
+        switch segcon.selectedSegmentIndex {
+        case 0:
+            self.configureHierarchy()
+            self.configureDataSource()
+        case 1:
+            self.createGridCollectionView()
+            self.configureGridDataSource()
+        default: return
+        }
```

Comment:
index로 접근하지말고 enum으로 관리하면 어떨까요?

```suggestion
eum SegmentItem: Int, CaseIterable {

    case list
    case grid
}

guard let selection = SegmentItem(rawValue: segcon.selectedSegmentIndex) else {
    return
}

switch segmentItem {
~~~
}
```

Reply:
segment enum으로 처리하도록 바꾸어 보았습니다!